.github/workflows/e2e-tests-main-devnet.yml

```diff
@@ -182,6 +182,25 @@ jobs:
           if-no-files-found: error
           retention-days: 7
 
+  run-aleph-client-subxt-codegen-check:
```

Comment:
When is it going to run? On nightly or every PR/commit?

Reply:
Every push to PR, and every merge to `main`
Latter is not necessarily required, our current pipelines works like this. Ideally we should separate build from e2e testing phase, meaning that
* every commit to PR should built test binary only and runs e2e tests (now it build both test and production binary)
* every merge PR to main should just build production binary and not run e2e tests

Reply:
This is a pretty have action, it involves:
* building the node binary (since it requires `run-e2e-test`)
* running a network
* pulling a docker image and running it

so it should not be run on every push to a PR - ideally it would be run only once before merging.

Reply:
This is not the scope of this PR I think.

Reply:
Can we move it to run on nightly then?

Reply:
It's not heavy action, it just reuses current pipelines. The check itself runs 4 minutes and it runs parallel to e2e tests that runs 15 minutes, so it does not extend overall pipeline time. 

When e2e are moved to nightly, we can move this check as welll.